tools/mo/openvino/tools/mo/utils/cli_parser.py

```diff
@@ -955,9 +955,16 @@ def validate_layout(layout: str):
     :param layout: string containing layout
     :raises: if layout is incorrect
     """
-    valid_layout_re = re.compile(r'\[?[^\[\]\(\)\s]*\]?')
-    if layout and not valid_layout_re.fullmatch(layout):
-        raise Error('Invalid layout parsed: {}'.format(layout))
+    error_msg = 'Invalid layout parsed: {}'.format(layout)
+    if layout:
+        incorrect_brackets = xor(layout[0] == '[', layout[-1] == ']')
```

Comment:
does `validate_layout` always receive layout with square brackets? Can it have circle brackets?

Reply:
`validate_layout` receive layout that should be later passed to preprocessing engine. It cannot have circle braces, but it can have no braces if the short form is provided, e.g. `nchw`. That is why I check if the square bracket is in the beginning it should be in the end. And contrary if it is not in the beginning it shouldn't be at the end.